### ISSUE TITLE
bug in trace_of_frobenius

### ISSUE THREAD

**Issue by AndrewVSutherland** [oscar-system/Oscar.jl#1480]
**Describe the bug**
For certain elliptic curves over finite fields the function `trace_of_frobenius` returns an incorect result

**To Reproduce**
Steps to reproduce the behavior, please provide a code snippet that triggers
the bug.
```
using Oscar
R,x = PolynomialRing(GF(59)); E = EllipticCurve(x^3+54*x+31,x)
trace_of_frobenius(E)
```
outputs -30, which is impossible,since |-30| > 2*sqrt(59)

**Expected behavior**
The correct output is 15 (as confirmed by Sage and Magma)

**System (please complete the following information):**
Please paste the output of `Oscar.versioninfo(full=true)` below. If this does
not work, please paste the output of Julia's `versioninfo()` and your Oscar
version.
```
OSCAR version 0.10.0
  combining:
    AbstractAlgebra.jl   v0.26.0
    GAP.jl               v0.8.1
    Hecke.jl             v0.14.4
    Nemo.jl              v0.31.1
    Polymake.jl          v0.8.0
    Singular.jl          v0.10.2
  building on:
    Antic_jll               v0.200.501+0
    Arb_jll                 v200.2200.0+0
    Calcium_jll             v0.400.102+0
    FLINT_jll               v200.800.500+0
    GAP_jll                 v400.1192.2+1
    Singular_jll            v403.1.300+0
    libpolymake_julia_jll   v0.8.0+2
    libsingular_julia_jll   v0.23.1+0
    msolve_jll              v0.2.3+1
    polymake_jll            v400.600.0+0
See `]st -m` for a full list of dependencies.

Julia Version 1.6.6
Commit b8708f954a (2022-03-28 07:17 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
Official https://julialang.org/ release
```

**Comment by thofma** [comment]
@JHanselman Can you have a look?

**Comment by thofma** [comment]
We just released 0.10.1 (available via `]up`), which fixes this.
